panic: interface conversion: interface {} is int, not string

## Description

On some machines Trivy is interrupting with the message
`panic: interface conversion: interface {} is int, not string`

## What did you expect to happen?
Trivy is running without error

## What happened instead?
The error occured and Trivy stops

## Output of run with `-debug`:

panic: interface conversion: interface {} is int, not string

goroutine 1 [running]:
github.com/aquasecurity/defsec/pkg/scanners/helm/parser.(*Parser).extractChartName(0xc0953c8c00, {0xc006be4500, 0x4c})
	/home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.68.3/pkg/scanners/helm/parser/parser.go:152 +0x494
github.com/aquasecurity/defsec/pkg/scanners/helm/parser.(*Parser).addPaths(0xc0953c8c00, {0xc08cd187c8?, 0x1, 0x3d3e840?})
	/home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.68.3/pkg/scanners/helm/parser/parser.go:120 +0xd5
github.com/aquasecurity/defsec/pkg/scanners/helm/parser.(*Parser).ParseFS.func1({0xc006be4500, 0x4c}, {0x7f5544a335d8, 0xc0953f01c0}, {0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.68.3/pkg/scanners/helm/parser/parser.go:105 +0x212
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc006be4500, 0x4c}, {0x7f5544a335d8, 0xc0953f01c0}, 0xc08cd189e8)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:66 +0x78
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc006be4460, 0x41}, {0x3d65f10, 0xc0953b3e60}, 0xc08cd189e8)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.WalkDir({0x3d3e840, 0xc02d8deed8}, {0xc006be4460, 0x41}, 0xc08cd189e8)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:112 +0xde
github.com/aquasecurity/defsec/pkg/scanners/helm/parser.(*Parser).ParseFS(0xc006be4460?, {0x3d63820?, 0xc002366660?}, {0x3d3e840?, 0xc02d8deed8?}, {0xc006be4460?, 0xc0953c8bf0?})
	/home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.68.3/pkg/scanners/helm/parser/parser.go:77 +0x93
github.com/aquasecurity/defsec/pkg/scanners/helm.(*Scanner).getScanResults(0xc0013470e0, {0xc006be4460, 0x41}, {0x3d63820, 0xc002366660}, {0x3d3e840?, 0xc02d8deed8})
	/home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.68.3/pkg/scanners/helm/scanner.go:138 +0xb8
github.com/aquasecurity/defsec/pkg/scanners/helm.(*Scanner).ScanFS.func1({0xc006be4460, 0x4c}, {0x7f5544a335d8, 0xc0953e1f40}, {0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.68.3/pkg/scanners/helm/scanner.go:119 +0x352
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc006be4460, 0x4c}, {0x7f5544a335d8, 0xc0953e1f40}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:66 +0x78
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc006be43c0, 0x41}, {0x7f5544a335d8, 0xc0953e1840}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc0953e8380, 0x32}, {0x7f5544a335d8, 0xc0953e17c0}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc0953e6420, 0x29}, {0x7f5544a335d8, 0xc0953e1740}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc0953e6390, 0x26}, {0x7f5544a335d8, 0xc0953e1440}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc0953e61e0, 0x21}, {0x7f5544a335d8, 0xc0953e0580}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc09504d920, 0x13}, {0x7f5544a335d8, 0xc088724f00}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc08877c480, 0xf}, {0x7f5544a335d8, 0xc088724ac0}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0xc08877c3f8, 0x4}, {0x7f5544a335d8, 0xc0887248c0}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.walkDir({0x3d3e840, 0xc02d8deed8}, {0x34b9aff, 0x1}, {0x3d65f10, 0xc088770ed0}, 0xc08cd19438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:85 +0x27d
io/fs.WalkDir({0x3d3e840, 0xc02d8deed8}, {0x34b9aff, 0x1}, 0xc002915438)
	/opt/hostedtoolcache/go/1.18.3/x64/src/io/fs/walk.go:112 +0xde
github.com/aquasecurity/defsec/pkg/scanners/helm.(*Scanner).ScanFS(0x3016ee0?, {0x3d63820?, 0xc002366660?}, {0x3d3e840?, 0xc02d8deed8?}, {0x34b9aff?, 0x1?})
	/home/runner/go/pkg/mod/github.com/aquasecurity/defsec@v0.68.3/pkg/scanners/helm/scanner.go:95 +0xac
github.com/aquasecurity/trivy/pkg/fanal/handler/misconf.misconfPostHandler.Handle({0xd8?}, {0x3d63820, 0xc002366660}, 0xc00007ac00?, 0xc00197cd20)
	/home/runner/work/trivy/trivy/pkg/fanal/handler/misconf/misconf.go:237 +0x446
github.com/aquasecurity/trivy/pkg/fanal/handler.Manager.PostHandle({{0xc00148c180?, 0x0?, 0x0?}}, {0x3d63820, 0xc002366660}, 0x4?, 0x7fff83ab2f5c?)
	/home/runner/work/trivy/trivy/pkg/fanal/handler/handler.go:74 +0xaf
github.com/aquasecurity/trivy/pkg/fanal/artifact/local.Artifact.Inspect({{0x7fff83ab2f5c, 0x1}, {0x7f558134bce8, 0xc0010dd860}, {{{0x0, 0x0, 0x0}, {0xc00148c200, 0x4, 0x4}}}, ...}, ...)
	/home/runner/work/trivy/trivy/pkg/fanal/artifact/local/fs.go:125 +0x474
github.com/aquasecurity/trivy/pkg/scanner.Scanner.ScanArtifact({{_, _}, {_, _}}, {_, _}, {{0xc000e15520, 0x2, 0x2}, {0xc000e15580, ...}, ...})
	/home/runner/work/trivy/trivy/pkg/scanner/scan.go:110 +0x103
github.com/aquasecurity/trivy/pkg/commands/artifact.scan({_, _}, {{0xc00131af00, 0xc00232a530, {0x3d30d48, 0x6}, 0x0, 0x0, {0xc000b21938, 0x12}}, ...}, ...)
	/home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:513 +0x3fe
github.com/aquasecurity/trivy/pkg/commands/artifact.(*runner).scanArtifact(_, {_, _}, {{0xc00131af00, 0xc00232a530, {0x3d30d48, 0x6}, 0x0, 0x0, {0xc000b21938, ...}}, ...}, ...)
	/home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:215 +0xc7
github.com/aquasecurity/trivy/pkg/commands/artifact.(*runner).scanFS(_, {_, _}, {{0xc00131af00, 0xc00232a530, {0x3d30d48, 0x6}, 0x0, 0x0, {0xc000b21938, ...}}, ...})
	/home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:201 +0xcd
github.com/aquasecurity/trivy/pkg/commands/artifact.(*runner).ScanRootfs(_, {_, _}, {{0xc00131af00, 0xc00232a530, {0x3d30d48, 0x6}, 0x0, 0x0, {0xc000b21938, ...}}, ...})
	/home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:188 +0x1d0
github.com/aquasecurity/trivy/pkg/commands/artifact.run({_, _}, {{0xc00131af00, 0xc00232a530, {0x3d30d48, 0x6}, 0x0, 0x0, {0xc000b21938, 0x12}}, ...}, ...)
	/home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:362 +0x6ce
github.com/aquasecurity/trivy/pkg/commands/artifact.Run(0xc00131af00, {0x34ca70b, 0x6})
	/home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:329 +0x168
github.com/aquasecurity/trivy/pkg/commands/artifact.RootfsRun(0xc00234f9e0?)
	/home/runner/work/trivy/trivy/pkg/commands/artifact/fs.go:37 +0x25
github.com/urfave/cli/v2.(*Command).Run(0xc00234f9e0, 0xc00131a4c0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/command.go:169 +0x631
github.com/urfave/cli/v2.(*App).RunContext(0xc000b03860, {0x3d637e8?, 0xc00012a020}, {0xc0001100d0, 0xd, 0xd})
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/app.go:341 +0xbc8
github.com/urfave/cli/v2.(*App).Run(...)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/app.go:247
main.main()
	/home/runner/work/trivy/trivy/cmd/trivy/main.go:16 +0x4f

## Output of `trivy -v`:

Version: 0.29.2
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-07-04 06:06:17.69639951 +0000 UTC
  NextUpdate: 2022-07-04 12:06:17.69639901 +0000 UTC
  DownloadedAt: 2022-07-04 11:30:28.872983088 +0000 UTC

## Additional details (base image name, container registry info...):
Please implement Trivy in that way, that if something weired is found, that it not stops working.
Please give also info which file triggers the crash.

--- a/go.mod
+++ b/go.mod
@@ -100,7 +100,7 @@ require (
 	github.com/alicebob/gopher-json v0.0.0-20200520072559-a9ecdc9d1d3a // indirect
 	github.com/apparentlymart/go-cidr v1.1.0 // indirect
 	github.com/apparentlymart/go-textseg/v13 v13.0.0 // indirect
-	github.com/aquasecurity/defsec v0.68.6
+	github.com/aquasecurity/defsec v0.68.10
 	github.com/asaskevich/govalidator v0.0.0-20200428143746-21a406dcc535 // indirect
 	github.com/aws/aws-sdk-go v1.44.46
 	github.com/beorn7/perks v1.0.1 // indirect
@@ -186,7 +186,7 @@ require (
 	github.com/knqyf263/nested v0.0.1
 	github.com/lann/builder v0.0.0-20180802200727-47ae307949d0 // indirect
 	github.com/lann/ps v0.0.0-20150810152359-62de8c46ede0 // indirect
-	github.com/liamg/iamgo v0.0.6 // indirect
+	github.com/liamg/iamgo v0.0.9 // indirect
 	github.com/liamg/jfather v0.0.7 // indirect
 	github.com/lib/pq v1.10.4 // indirect
 	github.com/liggitt/tabwriter v0.0.0-20181228230101-89fcab3d43de // indirect

--- a/integration/fs_test.go
+++ b/integration/fs_test.go
@@ -124,6 +124,14 @@ func TestFilesystem(t *testing.T) {
 			golden: "testdata/helm_testchart.json.golden",
 		},
 		{
+			name: "helm chart directory scanning with builtin policies and non string Chart name",
+			args: args{
+				securityChecks: "config",
+				input:          "testdata/fixtures/fs/helm_badname",
+			},
+			golden: "testdata/helm_badname.json.golden",
+		},
+		{
 			name: "secrets",
 			args: args{
 				securityChecks: "vuln,secret",

--- a/integration/testdata/fixtures/fs/helm_badname/Chart.yaml
+++ b/integration/testdata/fixtures/fs/helm_badname/Chart.yaml
@@ -1,0 +1,3 @@
+apiVersion: v2
+name: 1001
+version: 1.0.0

--- a/integration/testdata/helm_badname.json.golden
+++ b/integration/testdata/helm_badname.json.golden
@@ -1,0 +1,17 @@
+{
+  "SchemaVersion": 2,
+  "ArtifactName": "testdata/fixtures/fs/helm_badname",
+  "ArtifactType": "filesystem",
+  "Metadata": {
+    "ImageConfig": {
+      "architecture": "",
+      "created": "0001-01-01T00:00:00Z",
+      "os": "",
+      "rootfs": {
+        "type": "",
+        "diff_ids": null
+      },
+      "config": {}
+    }
+  }
+}

